src/plurals.rs

```diff
@@ -5,62 +5,203 @@ pub enum Resolver<'a> {
+        if let Some(i) = index_of(src, "?") {
+            if let Some(l) = index_of(src, ":") {
+                Ast::Ternary(
+                    Box::new(Ast::parse(&src[0..i])),
+                    Box::new(Ast::parse(&src[i + 1..l])),
+                    Box::new(Ast::parse(&src[l + 1..]))
+                )
+            } else {
+                panic!("Incorrect ternary expression, expected `:`")
+            }
+        } else {
+            Self::parse_ge(src)
+        }
+    }
+
+    fn parse_ge<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if let Some(i) = index_of(src, ">=") {
+            Ast::Op(">=", Box::new(Ast::parse(&src[0..i])), Box::new(Ast::parse(&src[i + 2..])))
+        } else {
+            Self::parse_gt(src)
+        }
+    }
+
+    fn parse_gt<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if let Some(i) = index_of(src, ">") {
+            Ast::Op(">", Box::new(Ast::parse(&src[0..i])), Box::new(Ast::parse(&src[i + 1..])))
+        } else {
+            Self::parse_le(src)
+        }
+    }
+
+    fn parse_le<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if let Some(i) = index_of(src, "<=") {
+            Ast::Op("<=", Box::new(Ast::parse(&src[0..i])), Box::new(Ast::parse(&src[i + 2..])))
+        } else {
+            Self::parse_lt(src)
+        }
+    }
+
+    fn parse_lt<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if let Some(i) = index_of(src, "<") {
+            Ast::Op("<", Box::new(Ast::parse(&src[0..i])), Box::new(Ast::parse(&src[i + 1..])))
+        } else {
+            Self::parse_eq(src)
+        }
+    }
+
+    fn parse_eq<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if let Some(i) = index_of(src, "==") {
+            Ast::Op("==", Box::new(Ast::parse(&src[0..i])), Box::new(Ast::parse(&src[i + 2..])))
+        } else {
+            Self::parse_neq(src)
+        }
+    }
+
+    fn parse_neq<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if let Some(i) = index_of(src, "!=") {
+            Ast::Op("!=", Box::new(Ast::parse(&src[0..i])), Box::new(Ast::parse(&src[i + 2..])))
+        } else {
+            Self::parse_mod(src)
+        }
+    }
+    fn parse_mod<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if let Some(i) = index_of(src, "%") {
+            Ast::Op("%", Box::new(Ast::parse(&src[0..i])), Box::new(Ast::parse(&src[i + 1..])))
+        } else {
+            Self::parse_not(src.trim())
+        }
+    }
+
+    fn parse_not<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if index_of(src, "!") == Some(0) {
+            Ast::Not(Box::new(Ast::parse(&src[1..])))
+        } else {
+            Self::parse_int(src.trim())
+        }
+    }
+
+    fn parse_int<'b, 'c>(src: &'b str) -> Ast<'c> {
+        u64::from_str_radix(src, 10)
+            .map(|x| Ast::Integer(x))
+            .unwrap_or_else(|_| Self::parse_n(src.trim()))
+    }
+
+    fn parse_n<'b, 'c>(src: &'b str) -> Ast<'c> {
+        if src == "n" {
+            Ast::N
+        } else {
+            panic!("Unexpected token: {}", src)
```

Comment:
Oof, let's not panic anywhere in the library. This probably means that we need to propagate a `Result` everywhere though.